src/gwt/src/org/rstudio/core/client/RegexUtil.java

```diff
@@ -67,6 +67,9 @@ private static final native String makeWordCharacter()
    public static final Pattern RE_SWEAVE_CHUNK_END =
          Pattern.create("^\\s*@\\s*$", "");
    
+   public static final Pattern RE_EMBEDDED_R_CHUNK_BEGIN =
```

Comment:
Is this intended to capture all valid knitr chunks in a file? If so it may not get indented chunks or chunks inside a blockquote (see knitr regex patterns and/or patterns we use in source capsule detection).

Reply:
It isn't; it is only used inside the embedded Ace instances (`EMBEDDED`) where we thankfully don't have to worry about that.